sdmetrics/reports/utils.py

```diff
@@ -479,6 +479,133 @@ def get_column_pair_plot(real_data, synthetic_data, column_names, metadata):
         return make_mixed_column_pair_plot(real_data, synthetic_data)
 
 
+def get_cardinality(parent_table, child_table, parent_primary_key, child_foreign_key):
+    """Return the cardinality of the parent-child relationship.
+
+    Args:
+        parent_table (pandas.DataFrame):
+            The parent table.
+        child_table (pandas.DataFrame):
+            The child table.
+        parent_primary_key (string):
+            The name of the primary key column in the parent table.
+        child_foreign_key (string):
+            The name of the foreign key column in the child table.
+
+    Returns:
+        pandas.DataFrame
+    """
+    child_counts = child_table[child_foreign_key].value_counts()
+    child_per_parent = child_counts.reindex(parent_table[parent_primary_key], fill_value=0)
+    child_per_parent = child_per_parent.reset_index()
+    child_per_parent.columns = [parent_primary_key, '# children']
+
+    cardinalities = child_per_parent.groupby('# children').size().reset_index(name='# parents')
+
+    return cardinalities.sort_values('# children')
+
+
+def generate_cardinality_plot(data, parent_primary_key, child_foreign_key):
+    """Generate a plot of the cardinality of the parent-child relationship.
+
+    Args:
+        data (pandas.DataFrame):
+            The cardinality data.
+        parent_primary_key (string):
+            The name of the primary key column in the parent table.
+        child_foreign_key (string):
+            The name of the foreign key column in the child table.
+
+    Returns:
+        plotly.graph_objects._figure.Figure
+    """
+    fig = px.histogram(data,
+                       x='# children',
+                       y='# parents',
+                       color='data',
+                       barmode='group',
+                       color_discrete_sequence=[DATACEBO_DARK, DATACEBO_LIGHT],
+                       pattern_shape='data',
+                       pattern_shape_sequence=['', '/'],
+                       nbins=max(data['# children']) - min(data['# children']) + 1,
+                       histnorm='probability density')
```

Comment:
minor: can we start all the parameters a line below and not as tabbed over

Reply:
Yes, done in [76fcf86](https://github.com/sdv-dev/SDMetrics/pull/384/commits/76fcf862286df85acac9ee2977216ac5c72f5a30)